pwiz_tools/Skyline/Model/Lib/AlphaPeptDeep/AbstractDeepLibraryBuilder.cs

```diff
@@ -30,12 +30,156 @@
+        {
+            var key = accession.Split(':')[0];
+            if (!_predictionSupport.ContainsKey(key))
+            {
+                return false;
+            }
+
+            return _predictionSupport[key].RetentionTime;
+        }
+        public bool IsCcsSupportedMod(string accession)
+        {
+            var key = accession.Split(':')[0];
+            if (!_predictionSupport.ContainsKey(key))
+            {
+                return false;
+            }
+
+            return _predictionSupport[key].Ccs;
+        }
+
+
+        public bool PeptideHasOnlyMs2SupportedMod(string modifiedPeptide)
+        {
+            string[] peptideParts = modifiedPeptide.Split(new[]{'[',']',':'});
+            bool supported = true;
+            foreach (string part in peptideParts)
+            {
+                if (int.TryParse(part, out int intResult))
+                {
+                    supported = supported && IsMs2SupportedMod(part);
+                }
+            }
+
+            return supported;
+        }
+        public bool PeptideHasOnlyRtSupportedMod(string modifiedPeptide)
+        {
+            string[] peptideParts = modifiedPeptide.Split(new [] { '[', ']', ':' });
+            bool supported = true;
+            foreach (string part in peptideParts)
+            {
+                if (int.TryParse(part, out int intResult))
+                {
+                    supported = supported && IsRtSupportedMod(part);
+                }
+            }
+
+            return supported;
+        }
+        public bool PeptideHasOnlyCcsSupportedMod(string modifiedPeptide)
+        {
+            string[] peptideParts = modifiedPeptide.Split(new [] { '[', ']', ':' });
+            bool supported = true;
+            foreach (string part in peptideParts)
+            {
+                if (int.TryParse(part, out int intResult))
+                {
+                    supported = supported && IsCcsSupportedMod(part);
+                }
+            }
+
+            return supported;
+        }
+    }
+
     /// <summary>
-    /// Abstract base class for AlphaPeptDeep and Carafe library builders so they can share code
+    /// Abstract base class for AlphaPeptDeep and Carafe library builders so they can share code. 
     /// </summary>
     public abstract class AbstractDeepLibraryBuilder : ILibraryBuildWarning
     {
-        public static IList<ModificationType> PopulateUniModList(IList<ModificationIndex> supportedList)
+        /// <summary>
+        /// Populate a list of ModificationTypes from those available in UniModData.
+        /// </summary>
+        /// <param name="supportedList">List of modifications.
+        /// - Empty list means *none* will be returned.
+        /// - Null list means *all* in UniModData will be returned.
+        /// </param>
+        /// <returns></returns>
+        public static IList<ModificationType> PopulateUniModList(IList<ModificationType> supportedList)
```

Comment:
I'm not sure I understand what this function does.
It takes in a list of ModificationType objects, and returns a different list of ModificationType objects.
What is the difference between the list that was passed in and the list that was returned?
Why does this transformation need to happen? What was wrong with the original list that was passed in that needed to be changed?